### ISSUE TITLE
[Bug Report] Defaults setting for `shell` does not take effect

### ISSUE THREAD

**Issue by qobilidop** [anishathalye/dotbot#129]
`conf_1.yaml`
```yaml
- defaults:
    shell:
      stdout: true

- shell:
  - echo "hello world"
```
gives
```
echo "hello world"
All commands have been executed

==> All tasks executed successfully
```

`conf_2.yaml`
```yaml
- shell:
  - 
    command: echo "hello world"
    stdout: true
```
gives
```
echo "hello world"
hello world
All commands have been executed

==> All tasks executed successfully
```

I expected them to give the same results, but the defaults setting for `shell` does not take effect.

The Dotbot version I've tested on is [`v1.11.0`](https://github.com/anishathalye/dotbot/releases/tag/v1.11.0).

**Comment by ooJerryLeeoo** [comment]
@qobilidop Try this:

```yaml
- defaults:
    shell:
      stdout: true

- shell:
  - 
    command: echo "hello world"
```

**Comment by qobilidop** [comment]
Thanks @ooJerryLeeoo , it works!

This is a good enough solution for me. But I think it might still be worth making `conf_1.yaml` an acceptable form?

**Comment by anishathalye** [comment]
Certainly a bug. Thanks for reporting it 😄 

Fixed in f1e829725506813276c0dff21f6e6bfff9873444!
